dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/oracle/AbstractOracleConnector.java

```diff
@@ -190,15 +176,46 @@ Properties buildProperties(@Nonnull ConnectorArguments arguments) {
   // jdbc:oracle:thin:<TNSName> (from 10.2.0.1.0)
   static String buildUrl(ConnectorArguments arguments) {
     String url = arguments.getUri();
-    if (url == null) {
-      String host = arguments.getHost();
-      int port = arguments.getPort(OPT_PORT_DEFAULT);
-      if (isOracleSid(arguments)) {
-        url = "jdbc:oracle:thin:@" + host + ":" + port + ":" + arguments.getOracleSID();
-      } else {
-        url = "jdbc:oracle:thin:@//" + host + ":" + port + "/" + arguments.getOracleServicename();
-      }
+    String host = arguments.getHost();
+    int port = arguments.getPort(OPT_PORT_DEFAULT);
+    if (url != null) {
+      checkNonUriFlags(arguments);
+      return url;
+    }
+    checkServiceAndSid(arguments);
+    if (arguments.getOracleSID() != null) {
+      return "jdbc:oracle:thin:@" + host + ":" + port + ":" + arguments.getOracleSID();
+    } else {
+      return "jdbc:oracle:thin:@//" + host + ":" + port + "/" + arguments.getOracleServicename();
+    }
+  }
+
+  private static void checkNonUriFlags(ConnectorArguments arguments) {
+    if (arguments.getHost() != null) {
+      throw extraFlagProvided("host");
+    } else if (arguments.getPort() != null) {
+      throw extraFlagProvided("port");
+    } else if (arguments.getOracleServicename() != null) {
+      throw extraFlagProvided("oracle-service");
+    } else if (arguments.getOracleSID() != null) {
+      throw extraFlagProvided("oracle-sid");
     }
-    return url;
+  }
+
+  private static void checkServiceAndSid(ConnectorArguments arguments) {
+    boolean hasService = arguments.getOracleServicename() != null;
+    boolean hasSid = arguments.getOracleSID() != null;
+    if (!hasService && !hasSid || hasService && hasSid) {
+      throw new MetadataDumperUsageException(
+          "Provide either -oracle-service or -oracle-sid for oracle dumper");
```

Comment:
The command-line options should be prefixed with double dash `--` not single dash:

```
$ dwh-migration-dumper 2>&1 | grep -i oracle
--oracle-service <String: ORCL>                            Service name for oracle                                   
--oracle-sid <String: orcl>                                SID name for oracle           
```

Also, I would suggest referencing the constants that contain flag names, so that it would be easier to find the usages later.

https://github.com/google/dwh-migration-tools/blob/3fdca7661eb364bbd9c0736852d4d65d40df6124/dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/ConnectorArguments.java#L112-L113

Reply:
Done